src/main/java/net/nicoulaj/maven/plugins/checksum/digest/CRC32FileDigester.java

```diff
@@ -41,42 +45,40 @@
     /**
      * Build a new instance of {@link CRC32FileDigester}.
      */
-    public CRC32FileDigester()
-    {
-        super( ALGORITHM );
+    public CRC32FileDigester() {
+        super(ALGORITHM);
     }
 
     /**
      * {@inheritDoc}
      */
-    public String calculate( File file )
-        throws DigesterException
-    {
+    public String calculate(File file, String salt)
+        throws DigesterException {
         CheckedInputStream cis;
-        try
-        {
-            cis = new CheckedInputStream( new FileInputStream( file ), new CRC32() );
-        }
-        catch ( FileNotFoundException e )
-        {
-            throw new DigesterException( "Unable to read " + file.getPath() + ": " + e.getMessage() );
+        try {
+            byte[] bytes = IOUtils.toByteArray(new FileInputStream(file));
```

Comment:
This will load the whole file in memory, also there are several unneeded array copies, I think this can be done without much overhead but just chaining the file input stream with a salt ByteArrayInputStream